src/pages/publication/contributors_tab/Contributor.tsx

```diff
@@ -18,55 +18,56 @@ const StyledInstitutionSelect = styled(ContributorStyles.Select)`
 
 interface ContributorProps {
   contributor: ContributorType;
-  dispatch: Dispatch<any>;
+  index: number;
+  setFieldValue: (name: string, value: any) => void;
+  swap: (indexA: number, indexB: number) => void;
+  remove: (index: number) => void;
 }
 
-const Contributor: React.FC<ContributorProps> = ({ contributor, dispatch }) => {
-  const handleCorrespondingAuthorChange = (contributor: ContributorType) => (event: React.ChangeEvent<any>) => {
-    contributor.corresponding = event.target.checked;
-    dispatch(updateContributor(contributor));
-  };
-
-  const handleInstitutionChange = (contributor: ContributorType) => (event: React.ChangeEvent<any>) => {
-    contributor.selectedInstitution = event.target.value;
-    dispatch(updateContributor(contributor));
-  };
-
-  const deleteContributor = (contributor: ContributorType): void => {
-    dispatch(removeContributor(contributor));
-  };
-
-  const onMoveContributor = (contributor: ContributorType, direction: number) => {
-    dispatch(moveContributor(contributor, direction));
-  };
-
+const Contributor: React.FC<ContributorProps> = ({ contributor, index, setFieldValue, swap, remove }) => {
   return (
-    <StyledContainer>
-      <ContributorStyles.PersonIcon />
-      <ContributorStyles.Name>{contributor.name}</ContributorStyles.Name>
-      <StyledInstitutionSelect
-        onChange={handleInstitutionChange(contributor)}
-        value={contributor.selectedInstitution || ''}
-        variant="outlined">
-        <MenuItem value="" key="-1" />
-        {contributor?.institutions?.map(institution => (
-          <MenuItem value={institution} key={institution}>
-            {institution}
-          </MenuItem>
-        ))}
-      </StyledInstitutionSelect>
-      <ContributorStyles.CorrespondingAuthor
-        onChange={handleCorrespondingAuthorChange(contributor)}
-        checked={contributor.corresponding}
-      />
-      <ContributorStyles.OrcidIcon>
-        {contributor.orcid && (
-          <img src="https://orcid.org/sites/default/files/images/orcid_24x24.png" alt="ORCID iD icon" />
-        )}
-      </ContributorStyles.OrcidIcon>
-      <ContributorStyles.ContributorsArrows contributor={contributor} onMoveContributor={onMoveContributor} />
-      <ContributorStyles.DeleteIcon onClick={() => deleteContributor(contributor)} />
-    </StyledContainer>
+    <Field name={`contributors.authors[${index}]`}>
+      {({ form: { values } }: any) => {
```

Comment:
verken form eller values ser ut til å brukes

Reply:
Brukes for å sette opp pilene for å flytte personer i lista

Reply:
ah, ser nå at `values` blir brukt lenger ned ja. ble lurt av git diff-en. Mulig å bruke `useFormikContext` her også da, fyi